scio-core/src/main/scala/com/spotify/scio/values/ApproxFilterSCollectionOps.scala

```diff
@@ -0,0 +1,93 @@
+package com.spotify.scio.values
+import com.google.common.hash.Funnel
+import com.spotify.scio.coders.Coder
+
+import scala.language.implicitConversions
+
+class ApproxFilterSCollectionOps[T: Coder](self: SCollection[T]) {
+
+  /**
+   * Converts the SCollection to an ApproxFilter using the specified Builder.
+   *
+   * Generic `to` for all [[ApproxFilter]] collections.
+   **/
+  def to[C[B] <: ApproxFilter[B]](
+    builder: ApproxFilterBuilder[T, C]
+  )(implicit coder: Coder[C[T]]): SCollection[C[T]] =
+    builder.build(self)
+
+  def toBloomFilter(
+    fpProb: Double
+  )(implicit f: Funnel[T], coder: Coder[BloomFilter[T]]): SCollection[BloomFilter[T]] =
```

Comment:
We should have our own well defined coder for `BloomFilter`, or rather, `Coder[ApproxFilter[_]]`, so shouldn't need this implicit? The `T` can be erased here since we don't need that for ser/de?

Reply:
Also do we want to avoid leaking `Funnel[T]` to users? Like what if we want to swap in a non-guava BF impl?

Reply:
This is a bit tricky.
For serialization we don't need the type `T`, however for deserialization we return a `BloomFilter[T]` for which need `T`.

Defining a `Coder[ApproxFilter[_]]` instead of `Coder[BloomFilter[_]]` means that the user won't be able extend `ApproxFilter` to add their own filters, since in that case they need to explicitly inject the coder which covers all implementation. To prevent this we can make ApproxFilter `sealed`, which is kind of restrictive. 

In order to hide the `Funnel[T]` we need to have a fallback for the `Funnel[T]`, and convey in some way that we are using something else as a fallback, which might be rare.
If we want to change the guava impl to something else, it might need a different Typeclass.
To abstract out of the type classes we have
```
  def to[C[B] <: ApproxFilter[B]](
    builder: ApproxFilterBuilder[T, C]
  )(implicit coder: Coder[C[T]]): SCollection[C[T]] =
```  
however to provide a shortcut for the user to direct go a BF, we have `BloomFilter` we have this method, hence this exposes the funnel.

I do think that we should hide the funnel from leaking, however we would need a fallback strategy.

Reply:
I changed this a bit. We can have a `Coder[ApproxFilter[T]]` as AprroxFilter are serializable and having the deserialization as a separate trait allows us to always have the Coder if an implicit `Deserializer` is present.